src/Address.php

```diff
@@ -31,6 +31,6 @@ public static function fromArray(array $data)
             'country' => null,
         ];
 
-        return new self(array_merge($defaults, array_intersect_key($data, $defaults)));
+        return new self(\array_filter(\array_merge($defaults, \array_intersect_key($data, $defaults))));
```

Comment:
This seems a rather "broad" way to approach it. There might be some cases where you'd want to update a value as empty, like `house_extension` or `street_address2`. Or will those be cleared if values are omitted?

Reply:
hmm valid point.

I see two options:
1. filter out fields selectively (from array) which can be ommited if null
2. use array filter with a closure which discards null